src/test/server/Server_test.cpp

```diff
@@ -367,11 +370,186 @@ class Server_test : public beast::unit_test::suite
         }
     }
 
+    /**
+     * @brief sink for writing all log messages to a stringstream
+     */
+    class CaptureSink : public beast::Journal::Sink
+    {
+        std::stringstream& strm_;
+    public:
+        CaptureSink(beast::severities::Severity threshold,
+            std::stringstream& strm)
+        : beast::Journal::Sink(threshold, false)
+        , strm_(strm)
+        {
+        }
+
+        void
+        write(beast::severities::Severity level, std::string const& text) override
+        {
+            strm_ << text;
+        }
+    };
+
+    /**
+     * @brief Log manager for CaptureSinks. This class holds the stream
+     * instance that is written to by the sinks. Upon destruction, all
+     * contents of the stream are assigned to the string specified in the
+     * ctor
+     */
+    class CaptureLogs : public Logs
+    {
+        std::stringstream strm_;
+        std::string& result_;
+
+    public:
+        CaptureLogs(std::string& result)
+            : Logs (beast::severities::kInfo)
+            , result_(result)
+        {
+        }
+
+        ~CaptureLogs() override
+        {
+            result_ = strm_.str();
+        }
+
+        std::unique_ptr<beast::Journal::Sink>
+        makeSink(std::string const& partition,
```

Comment:
yeah - it's part of the interface required of all `Logs` implementation..it's how different sinks are created for topics/partitions.